src/core/utils/Debounce/Debounce.test.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { Debounce } from '../Debounce/Debounce';
+import { HtmlInput } from '../../../reset/HtmlInput/HtmlInput';
+
+describe('debounce', () => {
+  it('runs the children function wrapped by it', () => {
+    const mockFunction = jest.fn(() => 12);
+    render(
+      <Debounce data-testid="abc" waitFor={1000}>
+        {mockFunction}
```

Comment:
This looks like it should always work regardless of context => useless test. The mockFunction is immediately called on render, so the wrapping it inside debounce does nothing. Rather put it so that some event triggers a callback that uses the debounce without delay, e.g. like below but without the waitFor.

Reply:
Replaced this with a test with no waitFor time.